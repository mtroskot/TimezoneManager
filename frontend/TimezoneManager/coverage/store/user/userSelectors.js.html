<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for store/user/userSelectors.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">store/user</a> userSelectors.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">35.71% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>5/14</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>5/10</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet"><a name='L1'></a><a href='#L1'>1</a>
<a name='L2'></a><a href='#L2'>2</a>
<a name='L3'></a><a href='#L3'>3</a>
<a name='L4'></a><a href='#L4'>4</a>
<a name='L5'></a><a href='#L5'>5</a>
<a name='L6'></a><a href='#L6'>6</a>
<a name='L7'></a><a href='#L7'>7</a>
<a name='L8'></a><a href='#L8'>8</a>
<a name='L9'></a><a href='#L9'>9</a>
<a name='L10'></a><a href='#L10'>10</a>
<a name='L11'></a><a href='#L11'>11</a>
<a name='L12'></a><a href='#L12'>12</a>
<a name='L13'></a><a href='#L13'>13</a>
<a name='L14'></a><a href='#L14'>14</a>
<a name='L15'></a><a href='#L15'>15</a>
<a name='L16'></a><a href='#L16'>16</a>
<a name='L17'></a><a href='#L17'>17</a>
<a name='L18'></a><a href='#L18'>18</a>
<a name='L19'></a><a href='#L19'>19</a>
<a name='L20'></a><a href='#L20'>20</a>
<a name='L21'></a><a href='#L21'>21</a></td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import { ADMIN, MANAGER, USER } from 'src/constants/userRoles';
import { createSelector } from 'reselect';
&nbsp;
export const userInfoSelector = <span class="fstat-no" title="function not covered" >st</span>ore =&gt; <span class="cstat-no" title="statement not covered" >store.user.user;</span>
export const userSelector = <span class="fstat-no" title="function not covered" >st</span>ore =&gt; <span class="cstat-no" title="statement not covered" >store.user;</span>
export const userRolesSelector = <span class="fstat-no" title="function not covered" >st</span>ore =&gt; <span class="cstat-no" title="statement not covered" >store.user.user.roles;</span>
export const accessTokenSelector = <span class="fstat-no" title="function not covered" >st</span>ore =&gt; <span class="cstat-no" title="statement not covered" >store.user.accessToken;</span>
&nbsp;
export const mainUserRoleSelector = createSelector(
  userRolesSelector,
<span class="fstat-no" title="function not covered" >  ro</span>les =&gt; {
<span class="cstat-no" title="statement not covered" >    if (roles.includes(ADMIN)) {</span>
<span class="cstat-no" title="statement not covered" >      return ADMIN;</span>
    } else <span class="cstat-no" title="statement not covered" >if (roles.includes(MANAGER)) {</span>
<span class="cstat-no" title="statement not covered" >      return MANAGER;</span>
    } else {
<span class="cstat-no" title="statement not covered" >      return USER;</span>
    }
  }
);
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Apr 01 2020 17:56:22 GMT+0200 (Central European Summer Time)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
<script src="../../block-navigation.js"></script>
</body>
</html>
