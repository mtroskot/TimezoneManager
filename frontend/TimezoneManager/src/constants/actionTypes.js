export const userActionTypes = Object.freeze({
  AUTH_AUTO_SIGN_IN: 'AUTH_AUTO_SIGN_IN',
  AUTH_USER: 'AUTH_USER',
  AUTH_USER_SUCCESS: 'AUTH_USER_SUCCESS',
  REGISTER_USER: 'REGISTER_USER',
  REGISTER_USER_SUCCESS: 'REGISTER_USER_SUCCESS',
  UPDATE_USER_INFO: 'UPDATE_USER_INFO',
  UPDATE_USER_INFO_SUCCESS: 'UPDATE_USER_INFO_SUCCESS',
  DELETE_USER: 'DELETE_USER',
  DELETE_USER_SUCCESS: 'DELETE_USER_SUCCESS',
  LOGOUT: 'LOGOUT',
  LOGOUT_SUCCESS: 'LOGOUT_SUCCESS',
  CHANGE_USER_ROLE: 'CHANGE_USER_ROLE',
  CHANGE_USER_ROLE_SUCCESS: 'CHANGE_USER_ROLE_SUCCESS'
});

export const uiActionTypes = Object.freeze({
  START_ACTION: 'START_ACTION',
  STOP_ACTION: 'STOP_ACTION',
  REFRESH_ACTION_START: 'REFRESH_ACTION_START',
  REFRESH_ACTION_STOP: 'REFRESH_ACTION_STOP',
  TOGGLE_POPUP_MESSAGE: 'TOGGLE_POPUP_MESSAGE',
  ADD_ERROR_ACTION: 'ADD_ERROR_ACTION',
  CLEAR_ERROR_ACTIONS: 'CLEAR_ERROR_ACTIONS'
});

export const timezoneActionTypes = Object.freeze({
  ADD_NEW_TIMEZONE_ENTRY: 'ADD_NEW_TIMEZONE_ENTRY',
  ADD_NEW_TIMEZONE_ENTRY_SUCCESS: 'ADD_NEW_TIMEZONE_ENTRY_SUCCESS',
  UPDATE_TIMEZONE_ENTRY: 'UPDATE_TIMEZONE_ENTRY',
  UPDATE_TIMEZONE_ENTRY_SUCCESS: 'UPDATE_TIMEZONE_ENTRY_SUCCESS',
  DELETE_TIMEZONE_ENTRY: 'DELETE_TIMEZONE_ENTRY',
  DELETE_TIMEZONE_ENTRY_SUCCESS: 'DELETE_TIMEZONE_ENTRY_SUCCESS',
  FETCH_TIMEZONE_ENTRIES: 'FETCH_TIMEZONE_ENTRIES',
  FETCH_TIMEZONE_ENTRIES_SUCCESS: 'FETCH_TIMEZONE_ENTRIES_SUCCESS'
});

export const searchActionTypes = Object.freeze({
  SEARCH_TIMEZONE_ENTRIES: 'SEARCH_TIMEZONE_ENTRIES',
  SEARCH_TIMEZONE_ENTRIES_SUCCESS: 'SEARCH_TIMEZONE_ENTRIES_SUCCESS',
  SEARCH_USERS: 'SEARCH_USERS',
  SEARCH_USERS_SUCCESS: 'SEARCH_USERS_SUCCESS',
  CLEAR_SEARCH: 'CLEAR_SEARCH'
});
