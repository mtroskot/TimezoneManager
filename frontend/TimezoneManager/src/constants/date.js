export const clockFormat = 'HH:mm:ss';
export const dateFormat = 'E, MMMM d';
