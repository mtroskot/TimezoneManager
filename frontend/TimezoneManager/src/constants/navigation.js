export const screenNames = Object.freeze({
  CLOCK: 'CLOCK',
  ADD_NEW_TIMEZONE: 'ADD_NEW_TIMEZONE',
  TIMEZONE_EDIT: 'TIMEZONE_EDIT',
  SEARCH: 'SEARCH',
  AUTH: 'AUTH',
  AUTH_EDIT: 'AUTH_EDIT'
});

export const stackNames = Object.freeze({
  APP_STACK: 'APP_STACK',
  CLOCK_STACK: 'CLOCK_STACK',
  AUTH_STACK: 'AUTH_STACK'
});
