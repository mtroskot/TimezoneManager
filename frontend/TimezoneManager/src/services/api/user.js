const login = (emailAddress, password) => ({
  url: `users/login`,
  options: {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json'
    },
    validateStatus(status) {
      return status === 200 || status === 400; // Accept only status code 200
    },
    data: { emailAddress, password }
  }
});

const register = (firstName, lastName, emailAddress, password) => ({
  url: `users/register`,
  options: {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json'
    },
    data: {
      firstName,
      lastName,
      emailAddress,
      password
    }
  }
});

const getUserInfo = authToken => ({
  url: `/users/me`,
  options: {
    method: 'GET',
    headers: {
      Authorization: `Bearer ${authToken}`
    }
  }
});

export default {
  login,
  register,
  getUserInfo
};
