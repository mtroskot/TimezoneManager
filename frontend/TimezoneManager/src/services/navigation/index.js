import { NavigationActions, StackActions } from 'react-navigation';
import { DrawerActions } from 'react-navigation-drawer';

let navigator;

function setTopLevelNavigator(navigatorRef) {
  navigator = navigatorRef;
}

function addListener(event, callback) {
  navigator.addListener(event, callback);
}

function navigate(routeName, params) {
  navigator.dispatch(
    NavigationActions.navigate({
      routeName,
      params
    })
  );
}

/**
 * Add a route on the top of the stack, and navigate forward to it
 * @param routeName
 * @param params
 */
function push(routeName, params) {
  navigator.dispatch(
    StackActions.push({
      routeName,
      params
    })
  );
}

/**
 * Replace current state with a new state
 * @param routeName
 * @param params
 */
function reset(routeName, params) {
  const resetAction = StackActions.reset({
    index: 0,
    actions: [
      NavigationActions.navigate({
        routeName,
        params
      })
    ]
  });
  navigator.dispatch(resetAction);
}

/**
 * Navigate to the top route of the stack, dismissing all other routes
 */
function popToTop() {
  navigator.dispatch(StackActions.popToTop());
}

function pop(n = 1) {
  navigator.dispatch(StackActions.pop({ n }));
}

function goBack() {
  navigator.dispatch(NavigationActions.back());
}

function closeDrawer() {
  navigator.dispatch(DrawerActions.closeDrawer());
}

function openDrawer() {
  navigator.dispatch(DrawerActions.openDrawer());
}

// add other navigation functions that you need and export them
const NavigationService = {
  setTopLevelNavigator,
  addListener,
  navigate,
  push,
  reset,
  pop,
  popToTop,
  goBack,
  closeDrawer,
  openDrawer
};
export default NavigationService;
