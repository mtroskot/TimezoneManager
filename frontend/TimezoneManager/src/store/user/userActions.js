import { userActionTypes } from 'src/constants/actionTypes';

export function authenticateUser(emailAddress, password) {
  return {
    type: userActionTypes.AUTH_USER,
    payload: { emailAddress, password }
  };
}

export const authenticateUserSuccess = (user, accessToken, refreshToken) => {
  return {
    type: userActionTypes.AUTH_USER_SUCCESS,
    payload: { user, accessToken, refreshToken }
  };
};

export function registerUser(registerData) {
  return {
    type: userActionTypes.REGISTER_USER,
    payload: { registerData }
  };
}

export const logoutUser = () => ({
  type: userActionTypes.LOGOUT,
  payload: {}
});

export const logoutUserSuccess = () => ({
  type: userActionTypes.LOGOUT_SUCCESS,
  payload: {}
});

export const changeUserRole = role => ({
  type: userActionTypes.CHANGE_USER_ROLE,
  payload: { role }
});

export const changeUserRoleSuccess = roles => ({
  type: userActionTypes.CHANGE_USER_ROLE_SUCCESS,
  payload: { roles }
});

export const authAutoSignIn = navigation => ({
  type: userActionTypes.AUTH_AUTO_SIGN_IN,
  payload: { navigation }
});

export function updateUserInfo(updateUserInfoForm) {
  return {
    type: userActionTypes.UPDATE_USER_INFO,
    payload: { updateUserInfoForm }
  };
}

export function updateUserInfoSuccess(updatedUserInfo) {
  return {
    type: userActionTypes.UPDATE_USER_INFO_SUCCESS,
    payload: { updatedUserInfo }
  };
}

export function deleteUser(userId) {
  return {
    type: userActionTypes.DELETE_USER,
    payload: { userId }
  };
}

export function deleteUserSuccess(userId) {
  return {
    type: userActionTypes.DELETE_USER_SUCCESS,
    payload: { userId }
  };
}

export function refreshTokenSuccess(accessToken, refreshToken) {
  return {
    type: userActionTypes.REFRESH_TOKEN_SUCCESS,
    payload: { accessToken, refreshToken }
  };
}
