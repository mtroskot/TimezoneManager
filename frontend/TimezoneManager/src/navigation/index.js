import { createAppContainer } from 'react-navigation';
import { createStackNavigator } from 'react-navigation-stack';
import { createDrawerNavigator } from 'react-navigation-drawer';
import SideDrawer from 'src/screens/SideDrawer';
import { stackNames } from 'src/constants/navigation';
import AuthStack from 'src/navigation/AuthStack';
import ClockStack from 'src/navigation/ClockStack';

const AppStack = createStackNavigator(
  {
    [stackNames.AUTH_STACK]: AuthStack,
    [stackNames.CLOCK_STACK]: ClockStack
  },
  {
    initialRouteName: stackNames.CLOCK_STACK,
    headerMode: 'none'
  }
);

AppStack.navigationOptions = ({ navigation }) => {
  let drawerLockMode = 'unlocked';
  if (navigation.state.index > 0) {
    drawerLockMode = 'locked-closed';
  }
  return {
    drawerLockMode
  };
};

const Navigator = createDrawerNavigator(
  {
    [stackNames.APP_STACK]: AppStack
  },
  {
    contentComponent: SideDrawer
  }
);

export default createAppContainer(Navigator);
