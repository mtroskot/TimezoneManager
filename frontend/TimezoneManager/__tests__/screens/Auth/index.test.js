import React from 'react';
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import AuthScreen from 'src/screens/Auth';
import { screenNames } from 'src/constants/navigation';
import { userActionTypes } from 'src/constants/actionTypes';
import { authenticateUser, registerUser, updateUserInfo } from 'src/store/user/userActions';

describe('AuthScreen wrapper', () => {
  it('AuthScreen renders correctly, Login form shown, isLoading=false ', () => {
    const mockStore = configureMockStore();
    const store = mockStore({
      ui: { loader: { actions: [], refreshing: [] } }
    });
    const navigation = { state: { routeName: screenNames.AUTH }, getParam: () => ({}) };

    const wrapper = shallow(<AuthScreen store={store} navigation={navigation} />)
      .dive()
      .dive();
    expect(wrapper.find('SafeAreaView')).toHaveLength(1);
    expect(wrapper.find('Memo(KeyboardAvoidAndDismissView)')).toHaveLength(1);
    expect(wrapper.find('Memo(LoginForm)')).toHaveLength(1);
    expect(wrapper.find('Memo(RegisterForm)')).toHaveLength(0);
    expect(wrapper.find('Memo(CustomButton)')).toHaveLength(1);

    expect(wrapper.find('Memo(LoginForm)').prop('loginForm')).toEqual({ emailAddress: '', password: '' });
    expect(wrapper.find('Memo(LoginForm)').prop('errors')).toEqual({});
    expect(wrapper.find('Memo(LoginForm)').prop('isLoading')).toEqual(false);

    expect(wrapper.find('Text')).toHaveLength(1);
    expect(wrapper.find('Text').props().children).toEqual("Don't have an account?");
    expect(wrapper.find('Memo(CustomButton)').prop('text')).toEqual('Create Account');
    expect(wrapper).toMatchSnapshot();
  });

  it('AuthScreen renders correctly, Login form shown, isLoading=true ', () => {
    const mockStore = configureMockStore();
    const store = mockStore({
      ui: {
        loader: {
          actions: [{ name: userActionTypes.AUTH_USER }],
          refreshing: []
        }
      }
    });
    const navigation = { state: { routeName: screenNames.AUTH }, getParam: () => ({}) };

    const wrapper = shallow(<AuthScreen store={store} navigation={navigation} />)
      .dive()
      .dive();
    expect(wrapper.find('SafeAreaView')).toHaveLength(1);
    expect(wrapper.find('Memo(KeyboardAvoidAndDismissView)')).toHaveLength(1);
    expect(wrapper.find('Memo(LoginForm)')).toHaveLength(1);

    expect(wrapper.find('Memo(LoginForm)').prop('loginForm')).toEqual({ emailAddress: '', password: '' });
    expect(wrapper.find('Memo(LoginForm)').prop('errors')).toEqual({});
    expect(wrapper.find('Memo(LoginForm)').prop('isLoading')).toEqual(true);

    expect(wrapper.find('Memo(RegisterForm)')).toHaveLength(0);
    expect(wrapper.find('Memo(CustomButton)')).toHaveLength(0);
    expect(wrapper.find('Text')).toHaveLength(0);
  });

  it('AuthScreen renders correctly, Register form shown, isEdit=true ', () => {
    const mockStore = configureMockStore();
    const store = mockStore({
      ui: {
        loader: { actions: [], refreshing: [] }
      }
    });
    const navigation = {
      state: { routeName: screenNames.AUTH_EDIT },
      getParam: () => ({ id: 1, firstName: 'Marko', lastName: 'Troskot', emailAddress: 'marko@hotmail.com' })
    };

    const wrapper = shallow(<AuthScreen store={store} navigation={navigation} />)
      .dive()
      .dive();
    expect(wrapper.find('SafeAreaView')).toHaveLength(1);
    expect(wrapper.find('Memo(KeyboardAvoidAndDismissView)')).toHaveLength(1);
    expect(wrapper.find('Memo(LoginForm)')).toHaveLength(0);
    expect(wrapper.find('Memo(RegisterForm)')).toHaveLength(1);
    expect(wrapper.find('Memo(RegisterForm)').prop('registerForm')).toEqual({
      id: 1,
      firstName: 'Marko',
      lastName: 'Troskot',
      emailAddress: 'marko@hotmail.com',
      password: '',
      matchingPassword: ''
    });
    expect(wrapper.find('Memo(RegisterForm)').prop('headerText')).toEqual('Update user account');
    expect(wrapper.find('Memo(RegisterForm)').prop('submitButtonText')).toEqual('Update Account');
    expect(wrapper.find('Memo(RegisterForm)').prop('errors')).toEqual({});
    expect(wrapper.find('Memo(RegisterForm)').prop('isEdit')).toEqual(true);
    expect(wrapper.find('Memo(RegisterForm)').prop('isLoading')).toEqual(false);

    expect(wrapper.find('Memo(CustomButton)')).toHaveLength(0);
    expect(wrapper.find('Text')).toHaveLength(0);
  });

  it('AuthScreen renders correctly, Register form shown, isEdit=true, isLoading=true ', () => {
    const mockStore = configureMockStore();
    const store = mockStore({
      ui: {
        loader: { actions: [{ name: userActionTypes.REGISTER_USER }], refreshing: [] }
      }
    });
    const navigation = {
      state: { routeName: screenNames.AUTH_EDIT },
      getParam: () => ({ id: 1, firstName: 'Marko', lastName: 'Troskot', emailAddress: 'marko@hotmail.com' })
    };

    const wrapper = shallow(<AuthScreen store={store} navigation={navigation} />)
      .dive()
      .dive();
    expect(wrapper.find('SafeAreaView')).toHaveLength(1);
    expect(wrapper.find('Memo(KeyboardAvoidAndDismissView)')).toHaveLength(1);
    expect(wrapper.find('Memo(LoginForm)')).toHaveLength(0);
    expect(wrapper.find('Memo(RegisterForm)')).toHaveLength(1);
    expect(wrapper.find('Memo(RegisterForm)').prop('registerForm')).toEqual({
      id: 1,
      firstName: 'Marko',
      lastName: 'Troskot',
      emailAddress: 'marko@hotmail.com',
      password: '',
      matchingPassword: ''
    });
    expect(wrapper.find('Memo(RegisterForm)').prop('headerText')).toEqual('Update user account');
    expect(wrapper.find('Memo(RegisterForm)').prop('submitButtonText')).toEqual('Update Account');
    expect(wrapper.find('Memo(RegisterForm)').prop('errors')).toEqual({});
    expect(wrapper.find('Memo(RegisterForm)').prop('isEdit')).toEqual(true);
    expect(wrapper.find('Memo(RegisterForm)').prop('isLoading')).toEqual(true);

    expect(wrapper.find('Memo(CustomButton)')).toHaveLength(0);
    expect(wrapper.find('Text')).toHaveLength(0);
  });
});

describe('interaction', () => {
  const mockStore = configureMockStore();
  const store = mockStore({
    ui: {
      loader: { actions: [], refreshing: [] }
    },
    authenticateUser: jest.fn(),
    registerUser: jest.fn(),
    updateUserInfo: jest.fn()
  });

  beforeEach(() => {
    store.clearActions();
  });

  it('calling handleInput should update searchInput state, isEdit=false', () => {
    const navigation = {
      state: { routeName: screenNames.AUTH },
      getParam: () => ({})
    };
    const wrapper = shallow(<AuthScreen store={store} navigation={navigation} />)
      .dive()
      .dive();
    wrapper.find('Memo(LoginForm)').prop('handleInput')('abc', 'emailAddress');
    expect(wrapper.find('Memo(LoginForm)').prop('loginForm')).toEqual({ emailAddress: 'abc', password: '' });

    wrapper.find('Memo(LoginForm)').prop('handleInput')('abc', 'password');
    expect(wrapper.find('Memo(LoginForm)').prop('loginForm')).toEqual({ emailAddress: 'abc', password: 'abc' });

    expect(wrapper.find('Memo(LoginForm)')).toHaveLength(1);
    expect(wrapper.find('Memo(RegisterForm)')).toHaveLength(0);
  });

  it('click create account should open register form and clicking again should return loginForm, isEdit=false', () => {
    const navigation = {
      state: { routeName: screenNames.AUTH },
      getParam: () => ({})
    };
    const wrapper = shallow(<AuthScreen store={store} navigation={navigation} />)
      .dive()
      .dive();
    expect(wrapper.find('Memo(LoginForm)')).toHaveLength(1);
    expect(wrapper.find('Memo(RegisterForm)')).toHaveLength(0);

    wrapper.find('Memo(CustomButton)').prop('onPress')();

    expect(wrapper.find('Memo(LoginForm)')).toHaveLength(0);
    expect(wrapper.find('Memo(RegisterForm)')).toHaveLength(1);

    wrapper.find('Memo(CustomButton)').prop('onPress')();

    expect(wrapper.find('Memo(LoginForm)')).toHaveLength(1);
    expect(wrapper.find('Memo(RegisterForm)')).toHaveLength(0);
  });

  it('clicking login should fail, validation not passed, isEdit=false', () => {
    const navigation = {
      state: { routeName: screenNames.AUTH },
      getParam: () => ({})
    };
    const wrapper = shallow(<AuthScreen store={store} navigation={navigation} />)
      .dive()
      .dive();

    wrapper.find('Memo(LoginForm)').prop('handleLogin')();
    expect(wrapper.find('Memo(LoginForm)').prop('errors')).toEqual({
      emailAddress: { display: true, errorMessage: 'Email Address is invalid' },
      password: { display: true, errorMessage: 'Password is invalid' }
    });

    expect(store.getActions()).toEqual([]);
  });

  it('clicking register should fail, validation not passed, isEdit=false', () => {
    const navigation = {
      state: { routeName: screenNames.AUTH },
      getParam: () => ({})
    };
    const wrapper = shallow(<AuthScreen store={store} navigation={navigation} />)
      .dive()
      .dive();

    wrapper.find('Memo(CustomButton)').prop('onPress')();
    wrapper.find('Memo(RegisterForm)').prop('handleRegister')();
    expect(wrapper.find('Memo(RegisterForm)').prop('errors')).toEqual({
      firstName: { display: true, errorMessage: 'First Name is invalid' },
      lastName: { display: true, errorMessage: 'Last Name is invalid' },
      emailAddress: { display: true, errorMessage: 'Email Address is invalid' },
      password: { display: true, errorMessage: 'Password is invalid' },
      matchingPassword: { display: true, errorMessage: 'Matching Password is invalid' }
    });

    expect(store.getActions()).toEqual([]);
  });

  it('clicking login should dispatch authenticate action, validation passed, isEdit=false', () => {
    const navigation = {
      state: { routeName: screenNames.AUTH },
      getParam: () => ({})
    };
    const wrapper = shallow(<AuthScreen store={store} navigation={navigation} />)
      .dive()
      .dive();

    wrapper.find('Memo(LoginForm)').prop('handleInput')('marko@hotmail.com', 'emailAddress');
    wrapper.find('Memo(LoginForm)').prop('handleInput')('123456', 'password');
    wrapper.find('Memo(LoginForm)').prop('handleLogin')();
    expect(wrapper.find('Memo(LoginForm)').prop('errors')).toEqual({});

    expect(store.getActions()).toEqual([authenticateUser('marko@hotmail.com', '123456')]);
  });

  it('clicking register should dispatch registerUser action, validation passed, isEdit=false', () => {
    const navigation = {
      state: { routeName: screenNames.AUTH },
      getParam: () => ({})
    };
    const wrapper = shallow(<AuthScreen store={store} navigation={navigation} />)
      .dive()
      .dive();

    wrapper.find('Memo(CustomButton)').prop('onPress')();
    wrapper.find('Memo(RegisterForm)').prop('handleInput')('Marko', 'firstName');
    wrapper.find('Memo(RegisterForm)').prop('handleInput')('Troskot', 'lastName');
    wrapper.find('Memo(RegisterForm)').prop('handleInput')('marko@hotmail.com', 'emailAddress');
    wrapper.find('Memo(RegisterForm)').prop('handleInput')('123456', 'password');
    wrapper.find('Memo(RegisterForm)').prop('handleInput')('123456', 'matchingPassword');
    wrapper.find('Memo(RegisterForm)').prop('handleRegister')();
    expect(wrapper.find('Memo(RegisterForm)').prop('errors')).toEqual({});

    expect(store.getActions()).toEqual([
      registerUser({
        id: null,
        firstName: 'Marko',
        lastName: 'Troskot',
        emailAddress: 'marko@hotmail.com',
        password: '123456',
        matchingPassword: '123456'
      })
    ]);
  });

  it('clicking update userInfo should fail, validation not passed, isEdit=true', () => {
    const navigation = {
      state: { routeName: screenNames.AUTH_EDIT },
      getParam: () => ({ id: 1, firstName: 'Marko', lastName: 'Troskot', emailAddress: 'marko@hotmail.com' })
    };
    const wrapper = shallow(<AuthScreen store={store} navigation={navigation} />)
      .dive()
      .dive();

    wrapper.find('Memo(RegisterForm)').prop('handleInput')('', 'firstName');
    wrapper.find('Memo(RegisterForm)').prop('handleInput')('', 'lastName');
    wrapper.find('Memo(RegisterForm)').prop('handleInput')('', 'emailAddress');
    wrapper.find('Memo(RegisterForm)').prop('handleRegister')();
    expect(wrapper.find('Memo(RegisterForm)').prop('errors')).toEqual({
      firstName: { display: true, errorMessage: 'First Name is invalid' },
      lastName: { display: true, errorMessage: 'Last Name is invalid' },
      emailAddress: { display: true, errorMessage: 'Email Address is invalid' }
    });

    expect(store.getActions()).toEqual([]);
  });

  it('clicking update userInfo should dispatch changeUserInfo action, validation  passed, isEdit=true', () => {
    const navigation = {
      state: { routeName: screenNames.AUTH_EDIT },
      getParam: () => ({ id: 1, firstName: 'Marko', lastName: 'Troskot', emailAddress: 'marko@hotmail.com' })
    };
    const wrapper = shallow(<AuthScreen store={store} navigation={navigation} />)
      .dive()
      .dive();

    wrapper.find('Memo(RegisterForm)').prop('handleRegister')();
    expect(wrapper.find('Memo(RegisterForm)').prop('errors')).toEqual({});

    expect(store.getActions()).toEqual([
      updateUserInfo({
        id: 1,
        firstName: 'Marko',
        lastName: 'Troskot',
        emailAddress: 'marko@hotmail.com'
      })
    ]);
  });
});
