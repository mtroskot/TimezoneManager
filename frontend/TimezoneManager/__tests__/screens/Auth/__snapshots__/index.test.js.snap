// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthScreen wrapper AuthScreen renders correctly, Login form shown, isLoading=false  1`] = `
<SafeAreaView
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Memo(KeyboardAvoidAndDismissView)
    viewStyle={
      Object {
        "backgroundColor": "#F2F2F2",
        "flex": 1,
      }
    }
  >
    <Memo(LoginForm)
      errors={Object {}}
      handleInput={[Function]}
      handleLogin={[Function]}
      isLoading={false}
      loginForm={
        Object {
          "emailAddress": "",
          "password": "",
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "marginTop": 59.21052631578947,
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "Ubuntu-Regular",
            "fontSize": 27.63157894736842,
            "marginBottom": 9.868421052631579,
          }
        }
      >
        Don't have an account?
      </Text>
      <Memo(CustomButton)
        onPress={[Function]}
        text="Create Account"
        textStyle={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "Ubuntu-Medium",
              "fontSize": 31.57894736842105,
              "textAlign": "center",
            },
            Object {
              "color": "#EAA79E",
            },
          ]
        }
      />
    </View>
  </Memo(KeyboardAvoidAndDismissView)>
</SafeAreaView>
`;
